+++ nginx/auto/cc/name
@@ -10,7 +10,7 @@ if [ "$NGX_PLATFORM" != win32 ]; then
     ngx_feature_run=yes
     ngx_feature_incs=
     ngx_feature_path=
-    ngx_feature_libs=
+    ngx_feature_libs=-static
     ngx_feature_test=
     . auto/feature

+++ nginx/auto/include
@@ -27,7 +27,7 @@ int main(void) {
 END


-ngx_test="$CC -o $NGX_AUTOTEST $NGX_AUTOTEST.c"
+ngx_test="$CC -pie -o $NGX_AUTOTEST $NGX_AUTOTEST.c"

 eval "$ngx_test >> $NGX_AUTOCONF_ERR 2>&1"

+++ nginx/auto/os/linux
@@ -37,7 +37,8 @@ fi
 ngx_feature="epoll"
 ngx_feature_name="NGX_HAVE_EPOLL"
 ngx_feature_run=yes
-ngx_feature_incs="#include <sys/epoll.h>"
+ngx_feature_incs="#include <sys/epoll.h>
+                  #include <stddef.h>"
 ngx_feature_path=
 ngx_feature_libs=
 ngx_feature_test="int efd = 0;

+++ njs/auto/feature
@@ -36,7 +36,7 @@


 nxt_test="$CC $CFLAGS $NXT_CFLAGS $NXT_CC_OPT $NXT_TEST_CFLAGS \
-          $nxt_feature_inc_path -o $NXT_AUTOTEST $NXT_AUTOTEST.c \
+          $nxt_feature_inc_path -static -o $NXT_AUTOTEST $NXT_AUTOTEST.c \
           $NXT_LD_OPT $NXT_TEST_LIBS $nxt_feature_libs"

 # /bin/sh -c "(...)" is to intercept "Killed", "Abort trap",
