From b1eb43c55a69d57cef40784234a3ba6513322ce1 Mon Sep 17 00:00:00 2001
From: Issam Maghni <me@concati.me>
Date: Mon, 22 Jul 2019 13:32:53 -0400
Subject: [PATCH 1/4] fix warning dangling-else

Signed-off-by: Issam Maghni <me@concati.me>
---
 src/ctype/towctrans.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/ctype/towctrans.c b/src/ctype/towctrans.c
index 8f681018..bd0136dd 100644
--- a/src/ctype/towctrans.c
+++ b/src/ctype/towctrans.c
@@ -259,12 +259,14 @@ static wchar_t __towcase(wchar_t wc, int lower)
 	 || (unsigned)wc - 0xabc0 <= 0xfeff-0xabc0)
 		return wc;
 	/* special case because the diff between upper/lower is too big */
-	if (lower && (unsigned)wc - 0x10a0 < 0x2e)
+	if (lower && (unsigned)wc - 0x10a0 < 0x2e) {
 		if (wc>0x10c5 && wc != 0x10c7 && wc != 0x10cd) return wc;
 		else return wc + 0x2d00 - 0x10a0;
-	if (!lower && (unsigned)wc - 0x2d00 < 0x26)
+	}
+	if (!lower && (unsigned)wc - 0x2d00 < 0x26) {
 		if (wc>0x2d25 && wc != 0x2d27 && wc != 0x2d2d) return wc;
 		else return wc + 0x10a0 - 0x2d00;
+	}
 	if (lower && (unsigned)wc - 0x13a0 < 0x50)
 		return wc + 0xab70 - 0x13a0;
 	if (!lower && (unsigned)wc - 0xab70 < 0x50)
-- 
2.22.0

